### PR TITLE
Linode: Mark 'name' as required. Fixes #29785

### LINKED ISSUE #29785: linode module: "name" parameter required, but documentation says it isn't
From @skyfaller on 2016-06-06T11:12:57Z
##### ISSUE TYPE
- Documentation Report
##### COMPONENT NAME

linode module
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = 
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

Running Ansible from OS X 10.11.5
##### SUMMARY

The documentation for the linode module at http://docs.ansible.com/ansible/linode_module.html claims that the "name" parameter is not required, but I seem to be unable to successfully use the linode module without it, it says "name is required for active state".

(P.S. I don't actually understand what value the "name" parameter should have, but using my server's hostname "caprice" makes the playbook run fine. What is the purpose of the "name" parameter, and how is it different from the "linode_id" parameter?)
##### STEPS TO REPRODUCE

Here's a sample playbook named "reboot.yml":

```

---

- hosts: caprice
  tasks:
  - name: Reboot the server
    local_action:
      module: linode
      api_key: "{{ linode_api_key }}"
#      name: caprice
      linode_id: "{{ linode_id }}"
      state: restarted
```
##### EXPECTED RESULTS

I expected the playbook to run successfully without the "name" parameter.
##### ACTUAL RESULTS

```
Vin:ansible nelson$ ansible-playbook reboot.yml --ask-vault-pass -vvvv
No config file found; using defaults
Vault password: 
Loaded callback default of type stdout, v2.0

PLAYBOOK: reboot.yml ***********************************************************
1 plays in reboot.yml

PLAY [caprice] *****************************************************************

TASK [setup] *******************************************************************
<caprice> ESTABLISH SSH CONNECTION FOR USER: None
<caprice> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/nelson/.ansible/cp/ansible-ssh-%h-%p-%r caprice '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1465211261.42-104881048472456 `" && echo ansible-tmp-1465211261.42-104881048472456="` echo $HOME/.ansible/tmp/ansible-tmp-1465211261.42-104881048472456 `" ) && sleep 0'"'"''
<caprice> PUT /var/folders/wj/fj_s9pp157xb_c7hb_r91rtm0000gn/T/tmpjrQJWt TO /home/nelson/.ansible/tmp/ansible-tmp-1465211261.42-104881048472456/setup
<caprice> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/nelson/.ansible/cp/ansible-ssh-%h-%p-%r '[caprice]'
<caprice> ESTABLISH SSH CONNECTION FOR USER: None
<caprice> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o ConnectTimeout=10 -o ControlPath=/Users/nelson/.ansible/cp/ansible-ssh-%h-%p-%r -tt caprice '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/nelson/.ansible/tmp/ansible-tmp-1465211261.42-104881048472456/setup; rm -rf "/home/nelson/.ansible/tmp/ansible-tmp-1465211261.42-104881048472456/" > /dev/null 2>&1 && sleep 0'"'"''
ok: [caprice]

TASK [Reboot the server] *******************************************************
task path: /Users/nelson/Code/server_documents/ansible/reboot.yml:5
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: nelson
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1465211263.04-60988620546823 `" && echo ansible-tmp-1465211263.04-60988620546823="` echo $HOME/.ansible/tmp/ansible-tmp-1465211263.04-60988620546823 `" ) && sleep 0'
<localhost> PUT /var/folders/wj/fj_s9pp157xb_c7hb_r91rtm0000gn/T/tmpMBOJkL TO /Users/nelson/.ansible/tmp/ansible-tmp-1465211263.04-60988620546823/linode
<localhost> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /Users/nelson/.ansible/tmp/ansible-tmp-1465211263.04-60988620546823/linode; rm -rf "/Users/nelson/.ansible/tmp/ansible-tmp-1465211263.04-60988620546823/" > /dev/null 2>&1 && sleep 0'
fatal: [caprice -> localhost]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"api_key": "vMQG7JAhCOKxDkVogfBVMg6vMwxiow0P0Q2Pt4XSOb566Bvt6yKFFhuDyBzGYw6V", "datacenter": null, "distribution": null, "linode_id": 1814698, "name": null, "password": null, "payment_term": 1, "plan": null, "ssh_pub_key": null, "state": "restarted", "swap": 512, "wait": true, "wait_timeout": "300"}, "module_name": "linode"}, "msg": "name is required for active state"}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @reboot.retry

PLAY RECAP *********************************************************************
caprice                    : ok=1    changed=0    unreachable=0    failed=1   

Vin:ansible nelson$ 
```

Copied from original issue: ansible/ansible-modules-core#3873

### PR DESCRIPTION
**Pull request by decentral1se**
##### SUMMARY

A fix motivated by https://github.com/ansible/ansible/issues/29785. While I have noted in https://github.com/ansible/ansible/issues/44696#issuecomment-416072275 that different requests require different required paramters (overall, implementing the documentation change in that comment is probably the 'final' paramter requirements fix), the OP in #29785 appears to be saying that `name` is required to use the module in the first place.

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

linode

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (mark-name-required-#29785 f112e314b7) last updated 2018/08/27 00:25:34 (GMT +200)
  config file = None
  configured module search path = [u'/home/decentral1se/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/decentral1se/hobby/ansible/lib/ansible
  executable location = /home/decentral1se/hobby/ansible/bin/ansible
  python version = 2.7.13 (default, Nov 24 2017, 17:33:09) [GCC 6.3.0 20170516]
```

##### ADDITIONAL INFORMATION

In the running of the test, I can't seem to test against the output but have manually verified:

```
lib/ansible/module_utils/basic.py:2367: SystemExit
-------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------

{"msg": "missing required arguments: name", "failed": true, "invocation": {"module_args": {"payment_term": 1, "displaygroup": "", "state": "present", "wait_timeout": 300, "swap": 512, "watchdog": true, "wait": true}}}
```

```

```

### CODE REVIEW

**Comment by rwaweber** [comment]
Love the idea of some tests being added around this -- much appreciated!

Thoughts on adding these doc fixes: https://github.com/ansible/ansible/pull/44700/files to this pr too?

**Comment by decentral1se** [comment]
> Love the idea of some tests being added around this -- much appreciated!

:rocket: 

> Thoughts on adding these doc fixes: https://github.com/ansible/ansible/pull/44700/files to this pr too?

Done in f6f8892945ebbc019d736d31726740bf44ae0c85.

**Comment by rwaweber** [comment]
shipit -- @InTheCloudDan @zbal thoughts on this?

**Comment by rwaweber** [comment]
@s-hertel could I get a gander/review on this?

@lwm is putting in a lot of great work on this module and I'd hate for them to get discouraged by having to wait for reviews from maintainers that don't join the conversation very often.

**Comment by InTheCloudDan** [comment]
shipit -- @rwaweber hit me up on IRC if things are popping up that I missed. 

I'm pretty much in same boat now haven't been using Linode that much myself.

**Review by achinthagunasekara — APPROVED** [review]

**Comment by resmo** [comment]
Thanks

**Pull request merged**
